### PR TITLE
White listed schemes

### PR DESCRIPTION
**Pull request by aaustin**
@derrickstaten @ahmednawar @sojanpr Urgent functionality exposure for premium partner.

This method allows the developer to white list URI schemes that Branch can track, in case there are private URI schemes that carry sensitive information.

### CODE REVIEW

File: Branch-SDK/Branch-SDK/Branch.m [hunk 1]
```diff
@@ -320,12 +322,28 @@ - (void) setDeepLinkDebugMode:(NSDictionary *)debugParams {
...
     BOOL handled = NO;
     if (url && ![url isEqual:[NSNull null]]) {
-        //always save the incoming url in the preferenceHelper in the externalIntentURI field
-        self.preferenceHelper.externalIntentURI = [url absoluteString];
+        
+        // save the incoming url in the preferenceHelper in the externalIntentURI field
+        if ([self.whiteListedSchemeList count]) {
+            for (NSString *scheme in self.whiteListedSchemeList) {
+                if ([scheme isEqualToString:[url scheme]]) {
+                    self.preferenceHelper.externalIntentURI = [url absoluteString];
```
Comment (sojanpr) [inline Branch-SDK/Branch-SDK/Branch.m]: Don't we need to `break` here

Comment (aaustin) [inline Branch-SDK/Branch-SDK/Branch.m]: sure good idea

**Comment by sojanpr** [comment]
LGTM 👍

**Comment by ahmednawar** [comment]
👍

**Pull request merged**